packages/components/src/toggle-group-control/toggle-group-control/as-radio-group.tsx

```diff
@@ -73,6 +73,13 @@ function UnforwardedToggleGroupControlAsRadioGroup(
 	const selectedValue = useStoreState( radio, 'value' );
 	const setValue = radio.setValue;
 
+	// Ensures that the active id is also reset after the value is "reset" by the consumer.
+	useEffect( () => {
+		if ( selectedValue === '' ) {
+			radio.setActiveId( undefined );
+		}
+	}, [ radio, selectedValue ] );
```

Comment:
I can feel the tech debt catching up to us 😅

Reply:
Food for thought: I wonder if this can be improved with a centralized controlled component layer that all controlled components would use. Probably a too idealistic idea, since some components use Ariakit and others do not, and this layer would be quite complicated to cover all cases.

Reply:
TBH I think that the most practical fix here is to change the APIs to follow the standard convention for controlled / uncontrolled component. Since that would be a breaking change, we could just release a V2 of the component that would give us also a chance to:

- implement / test [the new design spec](https://github.com/wordPress/gutenberg/issues/64439) alongside the legacy one
- review the current API strategy around "deselectability" and potentially multiple selection (should they be separate components? Or different subcomponents?)

Reply:
For this case specifically, I think there might be room to argue that resetting the active id on empty-ish (or invalid) values should be handled by Ariakit out of the box. I'll try and follow up.